mitodl/ocw-hugo-themes
# enhancement: upgrade hugo-bin-extended

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes https://github.com/mitodl/ocw-hugo-themes/issues/1161

#### What's this PR do?
Upgrades hugo-bin-extended

#### How should this be manually tested?

In this test we will test:
* the dev server is working correctly
* the local online and offline builds are working correctly 

1. Navigate to your local setup of ocw-hugo-themes.
2. Checkout `hussaintaj/1161-upgrade-hugo`
3. Run `yarn install`
4. Run `yarn start course` and `yarn start www`
5. **Expected Behavior** Both the `course` and `www` themes are functioning correctly for local development.
6. Run `yarn webpack:build:dev`
7. Grab a course (from the content repo), open it in a terminal, and run
    ```bash
    npx hugo-bin-extended@0.112.7 --themesDir path/to/ocw-hugo-themes/  --config path/to/ocw-hugo-projects/ocw-course-v2/config.yaml --baseURL /courses/site-id --destination output-online --buildDrafts
    ```
8. Then serve and test the build.
    ```bash
    cd output-online # or cd output-offline
    npx http-server
    ```
    **Expected Behavior**: The site works correctly. The URLs formed are in the format we defined using `--baseURL` `/course/site-id`, since we're statically serving these disk files, these URLs might return a 404. This is expected. Remove `course/site-id` and you'll find the content to view.
9. Repeat step 7 to 8 for the `www` site.
    ```bash
    npx hugo-bin-extended@0.112.7 --themesDir path/to/ocw-hugo-themes/  --config path/to/ocw-hugo-projects/ocw-www/config.yaml --destination output-online --buildDrafts
    ```
    Use the --baseURL option if you want to build the RC `ocw-www` that references other courses. Note that some URLs will break because of this, but you should be able to verify the functionality.
    ```bash
    npx hugo-bin-extended@0.112.7 --themesDir path/to/ocw-hugo-themes/  --config path/to/ocw-hugo-projects/ocw-www/config.yaml --baseURL https://live-qa.ocw.mit.edu/  --destination output-online --buildDrafts
    ```
10. Repeat step 7 to 9 for offline configs.
    ```bash
    npx hugo-bin-extended@0.112.7 --themesDir path/to/ocw-hugo-themes/  --config path/to/ocw-hugo-projects/ocw-course-v2/config-offline.yaml --baseURL /courses/site-id --destination output-offline --buildDrafts
    
    npx hugo-bin-extended@0.112.7 --themesDir path/to/ocw-hugo-themes/  --config path/to/ocw-hugo-projects/ocw-www/config-offline.yaml --destination output-offline --buildDrafts
    ```
11. Done. In this test, we have effectively tested the following features. Feel free to use the checklist below to track your progress.
     - [x] course dev server
     - [x] www dev server
     - [x] course online build
     - [x] www online build
     - [x] course offline build
     - [x] www offline build

#### Any background context you want to provide?
https://github.com/mitodl/ocw-studio/issues/1821
